pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -639,7 +673,7 @@ func shouldForceRemoval(job *batchv1.Job, d data.Object) bool {
 		return false
 	}
 
-	return condition.Cond("Failed").IsTrue(job)
+	return job.Labels[InfraJobRemove] == "true" && condition.Cond("Failed").IsTrue(job)
```

Comment:
Could you please explain why this additional condition was added? What specific case does it address?

Also, do we want to do `strings.ToLower` comparison here similar to how we do above for annotation just to be safe?

Reply:
Spoke offline about this. I will add the `strings.ToLower`.